validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -99,16 +102,35 @@ def broadcast_batch_by_batch_id_request(self, batch_id):
             batch_request,
             validator_pb2.Message.GOSSIP_BATCH_BY_BATCH_ID_REQUEST)
 
-    def broadcast(self, gossip_message, message_type):
+    def broadcast(self, gossip_message, message_type, exclude=None):
         # Gossip broadcasts are sent out in two different ways.
         # 1) To connected identities on our server socket
         # 2) To outbound connections we have originated
+        identities = copy.copy(self._identities)
+        connections = copy.copy(self._network.connections)
+
+        # Remove any identities that are in the exclude list
         for identity in self._identities:
+            if exclude is not None:
```

Comment:
This could defaulted at the start of the function to an empty list., should make this block, and the connection version a bit clearer.